pkg/controller/status/conditions_test.go

```diff
@@ -0,0 +1,468 @@
+package status
+
+import (
+	"reflect"
+	"testing"
+
+	configv1 "github.com/openshift/api/config/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func Test_conditions_entries(t *testing.T) {
+	time := metav1.Now()
+
+	type fields struct {
+		entryMap conditionsMap
+	}
+	tests := []struct {
+		name   string
+		fields fields
+		want   []configv1.ClusterOperatorStatusCondition
+	}{
+		{
+			name: "Can get the condition status from entry map",
+			fields: fields{entryMap: map[configv1.ClusterStatusConditionType]configv1.ClusterOperatorStatusCondition{
+				configv1.OperatorAvailable: {
+					Type:               configv1.OperatorAvailable,
+					Status:             configv1.ConditionUnknown,
+					LastTransitionTime: time,
+					Reason:             "",
+				},
+				configv1.OperatorProgressing: {
+					Type:               configv1.OperatorProgressing,
+					Status:             configv1.ConditionUnknown,
+					LastTransitionTime: time,
+					Reason:             "",
+				},
+			}},
+			want: []configv1.ClusterOperatorStatusCondition{
+				{
+					Type:               configv1.OperatorAvailable,
+					Status:             configv1.ConditionUnknown,
+					LastTransitionTime: time,
+					Reason:             "",
+				},
+				{
+					Type:               configv1.OperatorProgressing,
+					Status:             configv1.ConditionUnknown,
+					LastTransitionTime: time,
+					Reason:             "",
+				},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			c := &conditions{
+				entryMap: tt.fields.entryMap,
+			}
+			if got := c.entries(); !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("entries() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func Test_conditions_findCondition(t *testing.T) {
+	time := metav1.Now()
+
+	type fields struct {
+		entryMap conditionsMap
+	}
+	type args struct {
+		condition configv1.ClusterStatusConditionType
+	}
```

Comment:
this type is little bit redundant IMO

Reply:
Elaborate.

Reply:
I think it's not necessary to wrap `condition configv1.ClusterStatusConditionType` to a new type. IOW you can use directly the `configv1.ClusterStatusConditionType` IMO

Reply:
I see... I prefer to keep the `args` and `fields` wrapped. It makes the code more standardized test structures.

Reply:
`fields` and why pls? I can't see any benefit TBH. It adds only unnecessary parts to the test IMO. Using `conditions` you can omit the https://github.com/openshift/insights-operator/pull/498/files#diff-6cd7b045851c73c1120b823f3d07a763bfb7e20aa20e241ac5d3e27e862d57eaR119-R121 and just do `tt.conditions.findCondition(tt.args.condition)` then.